python/cudf/cudf/core/index.py

```diff
@@ -323,6 +323,74 @@ def _from_data(cls, data: MutableMapping, name: Any = no_default) -> Self:
     def _from_data_like_self(self, data: MutableMapping) -> Self:
         return _index_from_data(data, self.name)
 
+    @_performance_tracking
+    def to_pylibcudf(self, copy=False) -> tuple[plc.Column, dict]:
```

Comment:
docs failed saying `Index.to_pylibcudf` was missing. I went ahead and added both `to_plc` and `from_plc` in this PR. They are almost identical to `Series.from/to_plc`. In a follow up PR, I can centralize these methods since both `Index` and `Series` are `SingleColumnFrame`s